src/rust/ide/view/graph-editor/src/component/tooltip.rs

```diff
@@ -0,0 +1,224 @@
+//! The `Tooltip` shows extra information for UI components. It is pegged to the cursor location
+//! and appears when it receives information to show.
+
+use enso_frp as frp;
+use ensogl::application::Application;
+use ensogl::display;
+use ensogl::prelude::*;
+use ensogl::gui::style::*;
+use ensogl::define_style;
+
+use ensogl_gui_components::text_label::TextLabel;
+use ensogl::animation::hysteretic::HystereticAnimation;
+
+
+
+// =================
+// === Constants ===
+// =================
+
+const HIDE_DELAY_DURATION_MS : f32 = 150.0;
+const SHOW_DELAY_DURATION_MS : f32 = 150.0;
+
+
+
+// ==============
+// === Style ===
+// =============
+
+define_style! {
+    /// Host defines an object which the cursor position is bound to. It is used to implement
+    /// label selection. After setting the host to the label, cursor will not follow mouse anymore,
+    /// it will inherit its position from the label instead.
+    text   : Option<String>,
+}
+
+impl Style {
+    /// Create a `TooltipUpdate` that sets the label of the tooltip.
+    pub fn set_label(text:String) -> Self {
+        let text = Some(StyleValue::new(Some(text)));
+        Self{text}
+    }
+    /// Create a `TooltipUpdate` that unsets the label of the tooltip.
+    pub fn unset_label() -> Self {
+        let text = Some(StyleValue::new(None));
+        Self{text}
+    }
+
+    fn has_text(&self) -> bool {
+        if let Some(style_value) = self.text.as_ref() {
+            if let Some(inner) = style_value.value.as_ref() {
+                return inner.is_some()
+            }
+        }
+        false
+    }
+}
+
+
+
+// ==============
+// === Offset ===
+// ==============
+
+#[derive(Clone,Copy,Debug)]
+#[allow(missing_docs)]
+/// Indicates the position of the tooltip relative to the cursor location.
+pub enum Offset {
+    Top, Bottom, Left, Right
+}
+
+impl Default for Offset {
+    fn default() -> Self {
+        Offset::Bottom
+    }
+}
+
+ensogl::define_endpoints! {
+    Input {
+        set_style    (Style),
+        set_location (Vector2),
+        set_offset   (Offset)
```

Comment:
Next to placement I would also allow here to allow setting offset (from the cursor in pixels). If you could add it to the thee as well it will be perfect.